lib/Backend/Lower.cpp

```diff
@@ -8931,7 +8931,8 @@ Lowerer::LowerLdArrViewElem(IR::Instr * instr)
         }
         done = instr;
     }
-    InsertMove(dst, src1, done);
+    // Wasm memory buffer is not recycler allocated, so we shouldn't generate write barrier 
+    InsertMove(dst, src1, done, false);
```

Comment:
This is only for asm.js.
The right one in wasm is `Lowerer::LowerLdArrViewElemWasm`